### DISCUSSION TITLE
Support formatting TSDoc and JSDoc

### DISCUSSION THREAD

**Discussion by mctrafik** [biomejs/biome#2175] [Ideas]
I'd like to move off of prettier, but it does a great job with this plugin:

https://github.com/hosseinmd/prettier-plugin-jsdoc

to keep our documentation looking all nice. I.e. break at line-width, avoid duplicate spaces, put @param and @returns in the correct order etc. It's been invaluable, and for now preventing us from using biome to format.

**Comment by vaquilina-work** [comment]
+1

Our codebase uses TS extensions to JSDoc extensively, as per [this document.](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#type)

**Comment by xeho91** [comment]
Duplicate of https://github.com/biomejs/biome/discussions/741

**Comment by ematipico** [comment]
Duplicate of https://github.com/biomejs/biome/discussions/741
